python/tvm/micro/project.py

```diff
@@ -101,7 +101,26 @@ def __init__(self, api_client):
         if not self._info["is_template"]:
             raise NotATemplateProjectError()
 
+    def _check_project_options(self, options: dict):
```

Comment:
I'm wondering if it would be possible and better to rewrite that as:

```
available_options = [option["name"] for option in template.info()["project_options"]]
if not set(options.keys()).issubset(available_options):
   raise ValueError(...)
```

Reply:
done.